src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -166,9 +166,9 @@ public void Save(ModelSaveContext ctx)
 
         private static string TestType(ColumnType type)
         {
-            if (type.ItemType.IsText || type.ItemType.IsKey)
+            if (type.ItemType.IsText || type.ItemType.IsKey || type.ItemType == NumberType.R4 || type.ItemType == NumberType.R8)
                 return null;
-            return "Expected Text or Key item type";
+            return "Expected Text, Key, R4 or R8 item type";
```

Comment:
I would guess that most people don't know what R4 and R8 are for. `Single` and `Double` would be more recognizable to .NET users, and I believe is the recommended use for user text involving these types.